AsyncDisplayKit/Layout/ASStackLayoutSpec.mm

```diff
@@ -23,37 +23,104 @@
 
 @implementation ASStackLayoutSpec
 {
-  ASStackLayoutSpecStyle _style;
   std::vector<id<ASLayoutable>> _children;
 }
 
-+ (instancetype)newWithStyle:(ASStackLayoutSpecStyle)style children:(NSArray *)children
+- (instancetype)init
 {
-  ASStackLayoutSpec *spec = [super new];
-  if (spec) {
-    spec->_style = style;
-    spec->_children = std::vector<id<ASLayoutable>>();
+  return [self initWithDirection:ASStackLayoutDirectionHorizontal
+                         spacing:0.0
+            contentJustification:ASStackLayoutJustifyContentStart
+                   itemAlignment:ASStackLayoutAlignItemsStart
+                        children:nil];
+}
+
++ (instancetype)satckLayoutSpecWithDirection:(ASStackLayoutDirection)direction
+                                     spacing:(CGFloat)spacing
+                        contentJustification:(ASStackLayoutJustifyContent)justifyContent
+                               itemAlignment:(ASStackLayoutAlignItems)alignItems
+                                    children:(NSArray *)children
+{
+  return [[self alloc] initWithDirection:direction
```

Comment:
I think this would fit on one line nicely.  It feels harder to read this way.

Do you think the parallelism between contentJustification / justifyContent and itemAlignment / alignItems is correct?  I'm not sure I have a better idea, but because they are enum values and not BOOLs, I think both the method signature and argument should probably match in using contentJustification and itemAlignment.